matlab/schemas/+tuning/DotRF.m

```diff
@@ -0,0 +1,206 @@
+%{
+tuning.DotRF (computed) # receptive field from the dot stimulus
+-> preprocess.Sync
+-> preprocess.Spikes
+-> tuning.DotRFMethod
+-----
+%}
+
+classdef DotRF < dj.Relvar & dj.AutoPopulate
+    
+    properties
+        popRel = tuning.DotRFMethod * preprocess.Spikes & ...
+            (preprocess.Sync & (vis.ScanConditions & vis.SingleDot))
+    end
+    
+    methods(Access=protected)
+        
+        function makeTuples(self, key)
+            % get parameters
+            [mon_size, mon_dist] = fetch1(vis.Session &...
+                (preprocess.Spikes * preprocess.Sync & vis.SingleDot & key), ...
+                'monitor_size','monitor_distance');
+            [onset_delay, response_duration, rf_filter, shuffle, sd] = ...
+                fetch1(tuning.DotRFMethod & key, ...
+                'onset_delay', 'response_duration', 'rf_filter', 'shuffle', 'rf_sd');
+            
+            % get stimulus conditions
+            trials = fetch((preprocess.Sync & key) * vis.Trial * vis.SingleDot  ...
+                & 'trial_idx between first_trial and last_trial');
+            
+            conds = fetch(vis.Condition * vis.SingleDot & trials);
+            [locations_x,locations_y] = fetchn(vis.SingleDot & conds, 'dot_x', 'dot_y');
+            locations_x = unique(locations_x);
+            locations_y = unique(locations_y);
+            map_size = [length(locations_x) length(locations_y)];
+            deg2dot = 2*atand((mon_size/2)/(mon_dist/2.54))\norm(map_size);
+            
+            % get traces and frame times
+            [traces, trace_keys] = fetchn(preprocess.SpikesRateTrace & key, 'rate_trace');
+            traces = cell2mat(traces');
+            frame_times = fetch1(preprocess.Sync & key, 'frame_times');
+            frame_times = frame_times(1:size(traces,1));
+            
+            % get responses for each trial
+            response = nan(length(trials),size(traces,2));
+            index = nan(length(trials),1);
+            for itrial = 1:length(trials)
+                trial = trials(itrial);
+                trial.cond_idx = fetch1(vis.Trial & trial, 'cond_idx');
+                [loc_x, loc_y] = fetch1(vis.Condition * vis.SingleDot & trial, 'dot_x','dot_y');
```

Comment:
doing a bunch of fetch1's in a loop is inefficient.  You can do the same thing with something like
```matlab
trials = preprocess.Sync * vis.Trial * vis.SingleDot & ... 
    key & 'trial_idx_between first_trial and last_trial';
itrial = 0;
for trial = trials.fetch('cond_idx', 'dot_x', 'dot_y', 'flip_times')'
    itrial = itrial + itrial
    loc_x = trial.dot_x;
    loc_y = trial.dot_y;
    flip_times = trial.flip_times;
   ...
end